### ISSUE TITLE
[Core] Ray 1.12.0 fails on Google Colab

### ISSUE THREAD

**Issue by amogkam** [ray-project/ray#23951]
### What happened + What you expected to happen

Hi all,

We are aware of an issue with Ray 1.12.0 on Google Colab. Running `ray.init()` (or `tune.run` or executing any Ray code) will result in the following error

```
ValueError: invalid literal for int() with base 10: 'max\n'
```

This has been fixed in master, and the fix will be included in a 1.12.1 release which we will do shortly.

In the meantime, you can either use Ray master or use a version of Ray older than 1.12.0.

If you must use 1.12.0, you can execute the following cell prior to running Ray code to monkeypatch the system memory.

```
import psutil
import ray
ray._private.utils.get_system_memory = lambda: psutil.virtual_memory().total
```

### Versions / Dependencies

Ray 1.12.0

### Reproduction script

```python
import ray
ray.init()
```

**Comment by pranav-vempati** [comment]
Thanks for the suggested fix. What was the root cause of the issue?

**Comment by JessicaLopezEspejel** [comment]
Hello, 

I tried your solution. However, there is an error:

```
AttributeError: module 'pickle' has no attribute 'PickleBuffer'
```

![ray_error](https://user-images.githubusercontent.com/53188008/164415712-e33a9038-c5c1-4889-916d-fd47c2ae9927.PNG)

Do you have any advice? 
Thank you.

**Comment by JessicaLopezEspejel** [comment]
Hello, 

Your solution is working until I work with huggingface. It looks like the problem is in python 3.7, however  when I install python 3.8 in my colab:

```
!wget -O mini.sh https://repo.anaconda.com/miniconda/Miniconda3-py38_4.8.2-Linux-x86_64.sh
!chmod +x mini.sh
!bash ./mini.sh -b -f -p /usr/local
!conda install -q -y jupyter
!conda install -q -y google-colab -c conda-forge
!python -m ipykernel install --name "py38" --user
```
It does not recognize `transformers` , that I already install:

![ray_error](https://user-images.githubusercontent.com/53188008/164421186-365c14d0-a8f7-452f-95ce-ced284bec2a9.PNG)

It looks like the problem is in python 3.7, however
![python3 8_error](https://user-images.githubusercontent.com/53188008/164421737-e96d9e8b-bb67-49d7-b52a-3423f0c85494.PNG)

Thank you so much.

**Comment by amogkam** [comment]
cc @suquark any ideas on the pickle error?

**Comment by nayash** [comment]
I got this error while executing "tune.run" on Colab. Workaround which I did was installing v1.11.0 instead of 1.12.0

**Comment by NumberChiffre** [comment]
got the same error running the following in a local docker image built on my Mac, while I never seen this error on my Ubuntu machine, also python 3.17.13:

```python
import ray
ray.init()
```

**Comment by amogkam** [comment]
This has been fixed in Ray 1.12.1!
